quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes-deployment/pom.xml

```
@@ -89,4 +89,18 @@
     </plugins>
   </build>
 
+  <profiles>
+    <profile>
+      <id>productized</id>
+      <activation>
+        <property>
+          <name>productized</name>
+        </property>
+      </activation>
+      <properties>
+        <data-index-ephemeral.image>openshift-serverless-1/logic-data-index-ephemeral-rhel8:${project.version}</data-index-ephemeral.image>
```

I guess the image tag will not be `${project.version}`. Not sure what the tag will be. cc @AntStephenson

---

we might need to accept another parameter in build time that reflects the productized version. (with the `redhat-0001` suffix). If this is the `project.version`, than we should be good? @AntStephenson @spolti

---

What about the registry? is it set somewhere else?

---

The registry is Red Hat's isn't it? A machine with the configured registry would pull this image. But yeah, we also need to add it here.

---

When talking with @spolti earlier on GChat he suggested the value for this on the prod side should be `registry.redhat.io/openshift-serverless-1/logic-data-index-ephemeral-rhel8:1.24.0` noting there is no redhat-xxxxx on the version, should this be required I think this PR is sufficient (assuming we also add the registry) since the `${project.version}` is updated to reflect this.

---

Btw if it just needs to be 1.24.0 without the redhat version as @spolti wrote earlier, please just add a version property for it and we can take care of the update via some groovy. We are already storing `1.24.0` in our configs and using it for other purposes so it should be ok for us to extend and overwrite the new property.

---

oh brain fart from my side.

I don't think we need to include `redhat` suffix in image tags. 1.24.0, aka `{project.version}` should do. The only thing missing is the image registry and namespace.

---

Unfortunately in that case `${project.version}` will not work for prod, because this ends up becoming a productized version i.e 1.24.0.Final-redhat-00001. For us it would be preferred to have a separate version property that you could even default to `${project.version}` so we can nicely update it. 

Something like (with fixes for the registry):

```suggestion
        <data-index-ephemeral.image.tag>${project.version}</data-index-ephemeral.image.tag>
        <data-index-ephemeral.image>registry.redhat.io/openshift-serverless-1/logic-data-index-ephemeral-rhel8:${data-index-ephemeral.image.tag}</data-index-ephemeral.image>
```

That said, our ability to update this property is not contingent in it being split like above, I think it just makes it a bit easier for the version to be searched for and replaced. So if you prefer just having a singular property I'm fine with that too.

---

I'll update handling the image version, as for the registry, by adding it, dont we prevent users from using their own cache registry? If we hard code the registry users a locked into it. They should have that registry configured on their end, and in case they use like mirror/cache they wouldnt have that redhat registry.

---

ok, so we dont need to add the version in the property. We handle that in the code, also because of community. 
See https://github.com/kiegroup/kogito-runtimes/blob/main/quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes-deployment/src/main/java/org/kie/kogito/quarkus/processes/deployment/KogitoDevServicesProcessor.java#L109
That means we would end up with a tag of `1.24`, meaning we disregard the micro version. Would that be enough?
@AntStephenson @spolti

---

> I'll update handling the image version, as for the registry, by adding it, dont we prevent users from using their own cache registry? If we hard code the registry users a locked into it. They should have that registry configured on their end, and in case they use like mirror/cache they wouldnt have that redhat registry.

@cristianonicolai AFAIK If the user uses a mirror registry, it can configure the machine to use the mirror registry: https://docs.docker.com/registry/recipes/mirror/

---

I think disregarding the micro version should be enough here.

---

From the latest build we did the version was `1.24.0` I guess if it needs to be tagged with some alternative SPMM can probably help with that, do you think so @spolti  ?

---

@AntStephenson if we decide to use tags without the micro version as we do on Kogito, then yes, we would need to ask for this tag as well as we did recently with bamoe-kogito.

---

I would rather use the `Major.minor` only, that way we are sure to get latest bug fixes on the image

---

That sounds good to me, if all agrees, I'll create the SPMM request to do it.

---

+1

---

+1

---

btw, the request to retag is not done yet, isn't better wait for that?

---

@spolti, we're merging this on main only, I guess it will take some time before we run this for prod build.